packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -252,6 +254,24 @@ export class SharedTreeCore<
 		this.editManager.advanceMinimumSequenceNumber(brand(message.minimumSequenceNumber));
 	}
 
+	/**
+	 * Spawns a `SharedTreeBranch` that is based on the current state of the tree.
+	 * This can be used to support asynchronous checkouts of the tree.
+	 */
+	protected createBranch(): SharedTreeBranch<TChange> {
+		const branch = new SharedTreeBranch(
+			() => this.editManager.getLocalBranchHead(),
+			(forked) => {
+				const change = forked.pull();
+				this.submitEdit(this.editManager.fastForwardLocalBranch(forked.getHead()));
```

Comment:
`submitEdit` calls `EditManager.addLocalChange` which rebases the EditManager's AnchorSet over the new change, but it looks like we also rebase the AnchorSet in `fastForwardLocalBranch`, so we'd end up rebasing the anchors twice.